bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -385,53 +385,72 @@ private MiIoBasicChannel getChannel(String parameter) {
         return null;
     }
 
-    void updateProperties(MiIoSendCommand response) {
+    private void updatePropsFromJsonArray(MiIoSendCommand response) {
         JsonArray res = response.getResult().getAsJsonArray();
         JsonArray para = parser.parse(response.getCommandString()).getAsJsonObject().get("params").getAsJsonArray();
         if (res.size() != para.size()) {
             logger.debug("Unexpected size different. Request size {},  response size {}. (Req: {}, Resp:{})",
                     para.size(), res.size(), para.toString(), res.toString());
         }
         for (int i = 0; i < para.size(); i++) {
+            String param = para.get(i).getAsString();
             JsonElement val = res.get(i);
             if (val.isJsonNull()) {
-                logger.debug("Property '{}' returned null (is it supported?).", para.get(i).getAsString());
+                logger.debug("Property '{}' returned null (is it supported?).", param);
                 continue;
             }
-            MiIoBasicChannel basicChannel = getChannel(para.get(i).getAsString());
-            if (basicChannel != null) {
-                if (basicChannel.getTransfortmation() != null) {
-                    JsonElement transformed = Conversions.execute(basicChannel.getTransfortmation(), val);
-                    logger.debug("Transformed with '{}': {} {} -> {} ", basicChannel.getTransfortmation(),
-                            basicChannel.getFriendlyName(), val, transformed);
-                    val = transformed;
+            MiIoBasicChannel basicChannel = getChannel(param);
+            updateChannel(basicChannel, param, val);
+        }
+    }
+
+    private void updatePropsFromJsonObject(MiIoSendCommand response) {
+        JsonObject res = response.getResult().getAsJsonObject();
+        for (Object k : res.keySet()) {
+            String param = (String) k;
+            JsonElement val = res.get(param);
+            if (val.isJsonNull()) {
+                logger.debug("Property '{}' returned null (is it supported?).", param);
+                continue;
+            }
+            MiIoBasicChannel basicChannel = getChannel(param);
+            updateChannel(basicChannel, param, val);
+        }
+    }
+
+    private void updateChannel(MiIoBasicChannel basicChannel, String param, JsonElement val) {
+        if (basicChannel != null) {
+            if (basicChannel.getTransfortmation() != null) {
+                JsonElement transformed = Conversions.execute(basicChannel.getTransfortmation(), val);
+                logger.debug("Transformed with '{}': {} {} -> {} ", basicChannel.getTransfortmation(),
```

Comment:
Is that a typo? `getTransfor_t_mation`?

Reply:
It's a property in existing class MiIoBasicChannel.java, probably a typo. I don't think this PR is the right way to fix it.

Reply:
I can confirm its a typo :)